server/session/session.go

```diff
@@ -174,6 +175,22 @@ func (s service) RevokeSessionsForApp(r *incoming.Request, appID apps.AppID) err
 	return nil
 }
 
+func (s service) RevokeSessionsForAllUsers(r *incoming.Request, appID apps.AppID) error {
+	userKeys, err := s.store.Session.ListUsersWithSessions(appID)
+	if err != nil {
+		return err
+	}
+
+	for _, userKey := range userKeys {
+		err := s.RevokeSessionsForUser(r, userKey)
+		if err != nil {
+			r.Log.WithError(err).Warnw("failed to revoke session for user")
+		}
+	}
```

Comment:
This will cause #OfUsers RPC requests per app deactivation. That seems like a lot to me. Should the session revocation happen on the MM server side?

Reply:
yep, I was coming to that same conclusion, I'll check what can I do on the server to make this more performant

Reply:
Maybe it make even sense to add a `Disabled` state to the OAuth app server side and let the server handle the session revokation.

Reply:
@hanzei If we tell the server that this app is "disabled", we can then have the "soft revoke" the sessions. Meaning, if we re-enable the app then the sessions are back online. Just a thought. There are tradeoffs to that versus permanently revoking each session.

Reply:
Yep, that's what I also had in mind :+1: Let's traffic for the DB.

Reply:
👍

Reply:
> so even adding a disabled flag we couldn't (soft)revoke the sessions in the mattermost-server, we need the list of sessions stored in the plugin-apps KV store

Doesn't the server already have the list of session tied to an OAuth App via https://github.com/mattermost/mattermost-plugin-apps/blob/f05a93d06104115edc6ea816710974fcdbe2ed4a/server/session/session.go#L96-L103?

Reply:
I'm not sure, there is alot to cover to understand the session flow in the MM server. Thank you for the tip!